src/inmanta/server/services/orchestrationservice.py

```diff
@@ -134,77 +160,74 @@ def pair_resources_partial_update_to_old_version(
 
         for paired_resource in paired_resources:
             new_resource = paired_resource.new_resource
             old_resource = paired_resource.old_resource
-            assert new_resource is not None
             assert (
                 old_resource is None
-                or Id.parse_id(old_resource.resource["id"]).resource_str()
-                == Id.parse_id(new_resource.resource["id"]).resource_str()
+                or Id.parse_id(old_resource.resource.id).resource_str() == Id.parse_id(new_resource.resource.id).resource_str()
             )
             if paired_resource.is_new_resource():
-                merged_resources[new_resource.resource["id"]] = new_resource.resource
+                merged_resources[new_resource.resource.id] = new_resource.resource.dict()
             else:
                 if paired_resource.resource_changed_resource_set():
                     raise BadRequest(
-                        f"A partial compile cannot migrate resource {new_resource.resource['id']} to another resource set"
+                        f"A partial compile cannot migrate resource {new_resource.resource.id} to another resource set"
                     )
                 if new_resource.is_shared_resource() and paired_resource.is_update():
                     raise BadRequest(
-                        f"Resource ({new_resource.resource['id']}) without a resource set cannot"
+                        f"Resource ({new_resource.resource.id}) without a resource set cannot"
                         " be updated via a partial compile"
                     )
                 else:
-                    merged_resources[new_resource.resource["id"]] = new_resource.resource
+                    merged_resources[new_resource.resource.id] = new_resource.resource.dict()
         return list(merged_resources.values())
 
     def _merge_resource_sets(
         self, old_resource_sets: Dict[ResourceIdStr, Optional[str]], paired_resources: List[PairedResource]
     ) -> Dict[ResourceIdStr, Optional[str]]:
         updated_resource_sets: Set[str] = set(
-            res.new_resource.resource_set for res in paired_resources if res.new_resource.resource_set
+            res.new_resource.resource_set for res in paired_resources if res.new_resource.resource_set is not None
         )
-        changed_resource_sets: List[str] = list(updated_resource_sets) + self.removed_resource_sets
+        changed_resource_sets: Set[str] = set(list(updated_resource_sets) + list(self.removed_resource_sets))
         unchanged_resource_sets: Dict[ResourceIdStr, Optional[str]] = {
             k: v for k, v in old_resource_sets.items() if v not in changed_resource_sets
         }
         return {**unchanged_resource_sets, **self.resource_sets}
 
-    async def merge_partial_with_old(self) -> Tuple[List[object], Dict[ResourceIdStr, Optional[str]]]:
+    async def merge_partial_with_old(self) -> Tuple[List[Dict[str, Any]], Dict[ResourceIdStr, Optional[str]]]:
```

Comment:
Here too, why not stick with `object`? We really don't know anything more than that, do we?